### ISSUE TITLE
Add recipe for Code::Block

### ISSUE THREAD

**Issue by AwlsomeAlex** [haikuports/haikuports#2797]
Okay, so Code::Block will compile, however it is useless if I can't even get it to install. It might be something wrong to where I'm installing it, something with the File System, or something about the directories not existing. But I'm not that familiar with Haiku OS to know. Could someone assist me?
I executed `make install` and it failed at `/bin/install -c -m 644 SqPlusConst.h (rest of files..) '/usr/local/include/codeblocks/scripting/sqplus'`
![capture](https://user-images.githubusercontent.com/11639639/42611650-26e96178-8565-11e8-9c1a-13f1afec6286.PNG)

**Comment by waddlesplash** [comment]
Yep, we have a completely different directory structure vs. Linux. At this point it may be better if you try to write a recipe, as we have some tricks to take care of the directory differences in HaikuPorter...

**Comment by AwlsomeAlex** [comment]
Stinks Linux is all I know, I'm excited to expand my knowledge. I'll create the patch file and upload it and call it a day. Will probably attempt a recipe this weekend, because I hope to work on my Linux Distribution tomorrow. Might actually break the LSB and model it based off of Haiku in a way with packages, I like it and like being different.

**Comment by scottmc** [comment]
For learning about how haikuporter works you can start at the wiki page "Gentle intro to haikuporter", once you have it up and running and are ready to write your recipe(s), be sure to also read up on our development model.  We take new recipes and patches as pull requests.  Once a pull request is submitted it will be reviewed and corrections/updates/tips will be make and you can further improve on it.  Once it is acceptable we will likely merge it.

https://github.com/haikuports/haikuports/wiki/A-Gentle-Introduction-to-HaikuPorter-%231
https://github.com/haikuports/haikuports/wiki/Development

If you have any questions, feel free to ask here or on IRC #haiku on freenode, there's usually someone helpful there.

**Comment by extrowerk** [comment]
@AwlsomeAlex : it is a standard auto* package, so for testing you can reconfigure it like:
```configure --prefix=~/Desktop/codeblocks_test```
If you used extra configure switches you have to add them too.
Let it configure, rebuild it as you did before and use ```make install```. It will install the complete package into the ```~/Desktop/codeblocks_test``` folder.

From now you can start it. Be warned, if there is a ```lib``` folder in the resulting folder, you have to copy/symlink it to the folder, where the binary located (```./lib``` is always in LDPATH).

But yeah, a proper recipe would be nice.

**Comment by scottmc** [comment]
Let's keep this open, but make it an issue to Add a recipe for it.

**Comment by AwlsomeAlex** [comment]
Once I get a secondary device running I'll look more into this. (Virtualization just took way too long to compile. About 4 hours, which I don't have).

**Comment by Begasus** [comment]
Closing this one, a recipe has been in the repository for a long time, hence disabled.
Looked today at a recent build (25.03) that still builds with current patchset, launches (no crash so far), but not really production ready, can't seem to find plugins/compiler?

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/0ecfb7b2-6ff9-4a53-97c8-5a13d8149783" />
